rasa/nlu/selectors/response_selector.py

```diff
@@ -624,26 +656,66 @@ def _instantiate_model_class(self, model_data: RasaModelData) -> "RasaModel":
     @classmethod
     def load(
         cls,
-        meta: Dict[Text, Any],
-        model_dir: Text,
-        model_metadata: Metadata = None,
-        cached_component: Optional["ResponseSelector"] = None,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
         **kwargs: Any,
-    ) -> "ResponseSelector":
+    ) -> ResponseSelectorGraphComponent:
         """Loads the trained model from the provided directory."""
-        model: ResponseSelector = super().load(
-            meta, model_dir, model_metadata, cached_component, **kwargs
-        )
-        if not meta.get("file"):
-            return model
-
-        model.responses = meta.get("responses", {})
-        model.all_retrieval_intents = meta.get("all_retrieval_intents", [])
```

Comment:
why can't we just use the old approach and just load these 2 extra things from the model_storage?

Reply:
To double-check, you mean that I don't need to persist these 2 extra things separately and I can load them from `model`? I actually wasn't sure if the persisted model would contain both `responses` and `all_retrieval_intents`?

Reply:
No, I just meant using the old but with the changes `model_storage` changes and the other constructor call?

Reply:
Ahaa, took a while but I think I understand now your suggestion, it also removes a lot of duplicate code 👍🏼